control/src/lib.rs

```diff
@@ -133,117 +123,54 @@ pub mod pallet {
-	#[pallet::storage]
-	pub(super) type OrgConfiguration<T: Config> =
-		StorageMap<_, Blake2_128Concat, T::Hash, OrgConfig<T::Balance, T::CurrencyId>, OptionQuery>;
-
-	/// Org state (Inactive | Active | Locked) by org id.
-	///
-	/// OrgState: map Hash => ControlState
-	#[pallet::storage]
-	pub(super) type OrgState<T: Config> =
-		StorageMap<_, Blake2_128Concat, T::Hash, ControlState, ValueQuery, GetDefault>;
-
-	/// Org access model (Open | Voting | Controller) by org Hash (id).
-	///
-	/// OrgAccess: map Hash => AccessModel
-	#[pallet::storage]
-	pub(super) type OrgAccess<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, AccessModel, ValueQuery>;
+	pub(super) type OrgStates<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::Hash, OrgState, ValueQuery, GetDefault>;
 
 	/// Org members list by org id.
 	///
-	/// OrgMembers: map Hash => Vec<AccountId>
+	/// Members: map Hash => Vec<AccountId>
 	#[pallet::storage]
-	pub(super) type OrgMembers<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash,
-		BoundedVec<T::AccountId, T::MaxMembersPerOrg>, ValueQuery>;
+	pub(super) type Members<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash,
+		BoundedVec<T::AccountId, T::MaxMembers>, ValueQuery>;
 
 	/// Org members count by org id.
 	///
-	/// OrgMemberCount: map Hash => u64
+	/// OrgMemberCount: map Hash => MemberLimit
 	#[pallet::storage]
-	pub(super) type OrgMemberCount<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, u64, ValueQuery>;
+	pub(super) type OrgMemberCount<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, MemberLimit, ValueQuery>;
 
 	/// Member state (Inactive | Active ...) by org Hash and member account.
 	///
-	/// OrgMemberState: map (Hash, AccountId) => ControlMemberState
+	/// MemberStates: map Hash, AccountId => MemberState
 	#[pallet::storage]
-	pub(super) type OrgMemberState<T: Config> =
-		StorageMap<_, Blake2_128Concat, (T::Hash, T::AccountId), ControlMemberState, ValueQuery, GetDefault>;
+	pub(super) type MemberStates<T: Config> =
+	StorageDoubleMap<_, Blake2_128Concat, T::Hash, Blake2_128Concat, T::AccountId, MemberState, ValueQuery, GetDefault>;
 
-	/// Org list where account is a member.
+	/// Prime account of an Org.
 	///
-	/// Memberships: map AccountId => Vec<Hash>
+	/// OrgPrime: map Hash => AccountId
 	#[pallet::storage]
-	pub(super) type Memberships<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId,
-		BoundedVec<T::Hash, T::MaxOrgsPerAccount>, ValueQuery>;
-
-	/// Creator account of an Org.
-	///
-	/// OrgCreator: map Hash => AccountId
-	#[pallet::storage]
-	pub(super) type OrgCreator<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, T::AccountId, OptionQuery>;
-
-	/// Controller account of an Org.
-	///
-	/// OrgController: map Hash => AccountId
-	#[pallet::storage]
-	pub(super) type OrgController<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, T::AccountId, OptionQuery>;
+	pub(super) type OrgPrime<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, T::AccountId, OptionQuery>;
```

Comment:
Prime is already stored in the `Org` struct. Do we really need one more storage for this?

Reply:
Not really, I was also thinking about removing it. The idea was instead of reading the whole campaign, we could read only a prime. But I'm not so sure that we should do such read-optimizations, especially if it's not obvious that we need them.
I'd stick to the rule: simplify the code first and do optimizations later if needed. @2075 your opinion?
```
	fn org_controller_account(org_id: &T::Hash) -> Option<T::AccountId> {
		OrgPrime::<T>::get(org_id)
	}
```

Reply:
as I said before, these repetitive structures root in a pre graph stage of the protocol, there I optimised for fewer reads, now most of these are obsolete and can be efficiently packed like this:

- general struct for immutable, contractual data (e.g. org hash)
- meta struct for mutable, contractual data (e.g. cid for ifs)
- you may or may not need additional types / structs for certain maps

so if it doesn't complain, compress all into one/two structs and maps / double maps